modules/solid_mechanics/src/materials/CombinedNonlinearHardeningPlasticity.C

```diff
@@ -0,0 +1,241 @@
+#include "CombinedNonlinearHardeningPlasticity.h"
+#include "Function.h"
+#include "ElasticityTensorTools.h"
+#include "RadialReturnBackstressStressUpdateBase.h"
+#include "ElasticityTensorTools.h"
+
+registerMooseObject("SolidMechanicsApp", ADCombinedNonlinearHardeningPlasticity);
+registerMooseObject("SolidMechanicsApp", CombinedNonlinearHardeningPlasticity);
+
+template <bool is_ad>
+InputParameters
+CombinedNonlinearHardeningPlasticityTempl<is_ad>::validParams()
+{
+  InputParameters params = RadialReturnBackstressStressUpdateBaseTempl<is_ad>::validParams();
+  params.addClassDescription(
+      "This class uses the discrete material in a radial return combined"
+      "plasticity model. This class can simulate both nonlinear isotropic and kinematic"
+      " hardening effects.");
+  params.addParam<Real>("_q", 0.0, "Saturation value for isotropic hardening (Q in Voce model)");
```

Comment:
The parameter names shouldn't start with `_`, just the variable names in the C++.